### PR TITLE
Downloadable examples for “Run applications” section

### PR DESCRIPTION
**Pull request by sftim**
Issue #12740 is about making examples in the documentation downloadable, both for consistency & convenience.

This PR covers the examples within https://kubernetes.io/docs/tasks/run-application/

### CODE REVIEW

File: content/en/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 1]
```diff
@@ -65,18 +71,22 @@ Replication controller configuration files are described in
...
 
 ## Updating the container image
 
 To update only the container image, pass a new image name and tag with the
 `--image` flag and (optionally) a new controller name:
 
-    kubectl rolling-update NAME [NEW_NAME] --image=IMAGE:TAG
+```shell
+kubectl rolling-updatea NAME NEW_NAME --image=IMAGE:TAG
+```
```
Comment (kbhawkey) [inline content/en/docs/tasks/run-application/rolling-update-replication-controller.md]: - `kubectl rolling-update` is deprecated? Is there an alternative?
- check the spelling of rolling-update/rolling-updatea

Comment (sftim) [reply content/en/docs/tasks/run-application/rolling-update-replication-controller.md]: Good spot!
I might reword in a separate PR. I like PRs that do one thing.

**Comment by sftim** [comment]
Added 2 commits I'd forgotten to push

**Comment by tengqm** [comment]
/hold

This PR looks good. I'm leaving a hold here because for some unknown reasons our CI gate is not automatically triggered so we failed to validate the new YAML samples.

The CI gate was supposed to fail because this PR didn't change the `content/en/examples/examples_test.go` file, but it should.

**Comment by sftim** [comment]
Updating `content/en/examples/examples_test.go` - I should do that?

File: content/en/docs/tasks/run-application/configure-pdb.md [hunk 2]
```diff
@@ -135,31 +135,11 @@ You can find examples of pod disruption budgets defined below. They match pods w
...
-kind: PodDisruptionBudget
-metadata:
-  name: zk-pdb
-spec:
-  minAvailable: 2
-  selector:
-    matchLabels:
-      app: zookeeper
-```
+{{< codenew file="policy/zookeeper-pod-disruption-budget-minavailable.yaml" >}}
```
Comment (DanyC97) [inline content/en/docs/tasks/run-application/configure-pdb.md]: @sftim while i love the fact that you used the new shortcode, could you also add a note below or a sentence to say s'thing like _You can create it using `kubectl create -f https://`_ so folks new to docs can just copy the single line and off you go.

Of course those who are familiar with the docs can easily copy the url address from the block itself

File: content/en/docs/tasks/run-application/rolling-update-replication-controller.md [hunk 3]
```diff
@@ -143,24 +155,7 @@ from the [`kubectl` reference](/docs/reference/generated/kubectl/kubectl-command
...
-      labels:
-        app: nginx
-    spec:
-      containers:
-      - name: nginx
-        image: nginx:1.7.9
-        ports:
-        - containerPort: 80
-```
+{{< codenew file="controllers/replication-nginx-1.7.9.yaml" >}}
```
Comment (DanyC97) [inline content/en/docs/tasks/run-application/rolling-update-replication-controller.md]: same comment like https://github.com/kubernetes/website/pull/14147/files#r281527380

File: content/en/docs/tasks/run-application/configure-pdb.md (hunk 2 above)
Comment (DanyC97) [reply content/en/docs/tasks/run-application/configure-pdb.md]: of course, this assumes the example is already present in the examples dir, if not i'm as part of this PR we should add it

**Comment by DanyC97** [comment]
> Updating `content/en/examples/examples_test.go` - I should do that?

yes @sftim that is correct

**Comment by DanyC97** [comment]
Ref https://github.com/kubernetes/website/issues/12740 

thanks again @sftim for helping out !

**Comment by sftim** [comment]
Is this going to need marking `/lgtm` to run the updated code?
(if it doesn't - should it?)

**Comment by zacharysarah** [comment]
> (if it doesn't - should it?)

We don't have a ton of executable code in the repo, but still, we absolutely should have that kind of gate enabled. @spiffxp 👋 What steps do we need to take to enable `/ok-to-test` functionality for this repo?

**Comment by nikhita** [comment]
>  What steps do we need to take to enable /ok-to-test functionality for this repo?

`/ok-to-test` works only on prowjobs. If a non-member creates a PR, the `needs-ok-to-test` label will be added automatically and the prowjobs will start only when a member will add a `/ok-to-test` comment.

This repo doesn't seem to be having prowjobs, so enabling that option won't help. How/where are the tests for the executable code in this repo being run today?

**Comment by zacharysarah** [comment]
@nikhita 👋 
> /ok-to-test works only on prowjobs.

👍 

> How/where are the tests for the executable code in this repo being run today?

That's a highly germane question! We recently disabled Jenkins and have been relying on netlify to run example testing in the repo, specifically [this script](https://github.com/kubernetes/website/blob/master/scripts/test_examples.sh). Subsequently, as you can see earlier in the thread ☝️, @tengqm discovered that builds that should have failed are instead silently passing. 

/cc @lucperkins (who is out on vacation)

**Comment by sftim** [comment]
@tengqm is this OK to unhold now?

**Comment by tengqm** [comment]
/lgtm
/approve
Let's kick this in and fix the gate problem later in batch.

**Pull request merged**